src/daemon/delayed_shutdown_timer.cpp

```diff
@@ -48,19 +66,40 @@ void mp::DelayedShutdownTimer::start(std::chrono::milliseconds delay)
                  fmt::format("Shutdown request delayed for {} minute{}",
                              std::chrono::duration_cast<std::chrono::minutes>(delay).count(),
                              delay > std::chrono::minutes(1) ? "s" : ""));
-        QObject::connect(&shutdown_timer, &QTimer::timeout, [this]() { shutdown_instance(); });
+        ssh_session.exec(shutdown_message(std::chrono::duration_cast<std::chrono::minutes>(delay)));
+
+        time_remaining = delay;
+        std::chrono::minutes time_elapsed{1};
+        QObject::connect(&shutdown_timer, &QTimer::timeout, [this, delay, time_elapsed]() mutable {
+            time_remaining = delay - time_elapsed;
+            ssh_session.exec(shutdown_message(std::chrono::duration_cast<std::chrono::minutes>(time_remaining)));
+            if (time_elapsed >= delay)
+            {
+                shutdown_timer.stop();
+                shutdown_instance();
+            }
+            else
+            {
+                time_elapsed += std::chrono::minutes(1);
+            }
+        });
 
         virtual_machine->state = VirtualMachine::State::delayed_shutdown;
 
-        shutdown_timer.setSingleShot(true);
-        shutdown_timer.start(delay);
+        shutdown_timer.start(delay < std::chrono::minutes(1) ? delay : std::chrono::minutes(1));
```

Comment:
Shouldn't this be `delay <=`?

Reply:
I think `<=` will accomplish the same thing as `<` in this case since both result in 1 minute timers if `delay` is set to ` minute.

I basically have the check for `delay` for the tests so I don't think it's really important since the delayed shutdown command itself uses minute resolution for the dealy.

Reply:
Oh and yeah, DUH!